src/sc2laddercore/LadderManager.cpp

```diff
@@ -857,18 +861,30 @@ bool LadderManager::LoadSetup()
 	std::string EnableReplayUploadString = Config->GetValue("EnableReplayUpload");
 	if (EnableReplayUploadString == "True")
 	{
-		EnableReplayUploads = true;
+#ifndef ENABLE_CURL
+		throw std::logic_error(
+				"ERROR: Project was compiled without ENABLE_CURL but ladder manager is configured with EnableReplayUpload=True!"
```

Comment:
Probably better to drop the corresponding config option?
Exception is not a good way to report errors to the cli user.

Reply:
Same for server login.